spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -304,4 +304,9 @@ public boolean caseSensitive() {
         .defaultValue(SQLConf.CASE_SENSITIVE().defaultValueString())
         .parse();
   }
+
+  public String branch() {
+    return confParser.stringConf().option(SparkWriteOptions.BRANCH).parseOptional();
```

Comment:
Can this return main if no branch is passed in?

Reply:
Sure, this is still WIP. I wanted to know if i am missing some major operations

Reply:
sure just took a look at SparkWrite, I think we're missing the streaming append/overwrite cases?

https://github.com/apache/iceberg/blob/master/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java#L573

https://github.com/apache/iceberg/blob/master/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java#L591

Reply:
@amogh-jahagirdar Added tests